pkg/controller/provider/container/vsphere/collector.go

```diff
@@ -508,18 +510,18 @@ func (r *Collector) connect(ctx context.Context) error {
 	soapClient.SetThumbprint(url.Host, r.thumbprint())
 	vimClient, err := vim25.NewClient(ctx, soapClient)
 	if err != nil {
-		return liberr.Wrap(err)
+		return 0, liberr.Wrap(err)
 	}
 	r.client = &govmomi.Client{
 		SessionManager: session.NewManager(vimClient),
 		Client:         vimClient,
 	}
 	err = r.client.Login(ctx, url.User)
 	if err != nil {
-		return liberr.Wrap(err)
+		return http.StatusUnauthorized, liberr.Wrap(err)
```

Comment:
Is it certain that this is the only error we could get from Login? If the Login fails for other reasons (network?) then it might be misleading to turn that into a StatusUnauthorized.